core/src/test/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceBuilderTest.java

```diff
@@ -191,6 +191,59 @@ public void root(@Header("a") ArrayList<String> a,
                              @Header("a") LinkedList<String> b) {}
         });
 
+        // Multiple paths are allowed
+        Server.builder().annotatedService(new Object() {
+            @Get
+            @Post
+            @Path("/a")
+            @Path("/b")
+            public void root(BeanB b) {}
+        });
+
+        Server.builder().annotatedService(new Object() {
+            @Path("/")
+            @Get("/")
+            public void root() {}
```

Comment:
What happens in the following case?

```java
@Get("/a")
@Path("/b")
```

Does it have the same effect as:

```java
@Get
@Path("/a")
@Path("/b")
```

..?

Then, what happens in the following case?

```java
@Get("/a")
@Post("/b")
@Path("/c")
```

Is it same as:

```java
// Binding 1
@Get
@Path("/a")
@Path("/c")
// Binding 2
@Post
@Path("/b")
@Path("/c")
```

..?

Could you add the test cases for these cases in:

- `AnnotatedHttpServiceBuilderTest`
- `AnnotatedHttpServiceTest`

Alternatively, we could consider always rejecting the case where a method annotation with a path is used with a `Path` annotation. Which do you think is better, @minwoox @ikhoon @jrhee17 ?

Reply:
To reduce complication, I prefer to using a "/my/path" only one annotation. 😀  
- `@Get` and `@Path("/b")` 👍
- `@Get("/a")` and `@Path()` 👍
- `@Get("/a")` and `@Path("/b")` 👎

Reply:
> Alternatively, we could consider always rejecting the case where a method annotation with a path is used with a Path annotation

That's a great idea. That will make everything simple. To clarify:
```java
@Get("/a")   // Exception
@Path("/b")  // @Get should be alone or should not have "/a" when it's used with @Path
```
```java
@Get("/a")   // Exception
@Path        // Path should always have a path string
```
```java
@Get         // OK
@Path("/a")
@Path("/b")
```
```java
@Get         // OK
@Post
@Path("/c")
```
One thing, we have to consider is when only `@Path` is specified:
```java
@Path("/a")
```
Do we have to throw an exception or should allow all HTTP methods.
I slighly lean toward allowing all HTTP methods because we do that in route builder.

Reply:
> Could you add the test cases for these cases

roger that :D

>  consider always rejecting the case where a method annotation with a path is used with a Path annotation

+1 for me as well as it is visually easier to understand.
I was initially concerned there might be cases where users might want only a certain http method to have a path. In the long term, I think allowing multiple httpMethod patterns can help with this. (i.e. allowing multiple `@Get`, `@Post`, etc...)

> Do we have to throw an exception or should allow all HTTP methods.

+1 as it would also be consistent with other frameworks (like spring)

Reply:
> Do we have to throw an exception or should allow all HTTP methods. I slighly lean toward allowing all HTTP methods because we do that in route builder.
> +1 as it would also be consistent with other frameworks (like spring)

+1 for which one? I'm not sure which is better to be honest :-)

Reply:
just to clarify

```
@Get("/a")        // also OK
@Post("/b") 
@Path
```

> +1 for which one? I'm not sure which is better to be honest :-)

I meant the latter (allowing all http methods).

Looking at this again with a fresh mind, it does feel a little odd that the behavior of `@Path` changes depending on the existence of other annotations that aren't explicitly related.
🤔 🤔 🤔 🤔 🤔

Reply:
```java
@Get("/a")        // Not OK!
@Post("/b") 
@Path
```
I think this is not ok @jrhee17!
`@Path` should always have a path string and the above example does not make any different from:
```java
@Get("/a")  
@Post("/b") 
```

So I think we should raise an exception. WDYT?

Reply:
oops.  I meant the following. sry for the confusion :(
```
@Get("/a")  
@Post("/b") 
```

Reply:
yep:)  the test case has been moved.

I'm haven't added any special logic for the following case yet. (I don't think we reached a clear consensus)
```java
@Path("/")
public void pathByItself() {}
```

Reply:
Had a chat with @trustin and let's raise an exception in this case!